agent/consul/acl_endpoint_test.go

```diff
@@ -897,6 +897,32 @@ func TestACLEndpoint_TokenDelete(t *testing.T) {
 		assert.NoError(err)
 	}
 
+	// can't delete itself
```

Comment:
Given that this PR fixes both the self-delete issue and the segfault when deleting nonexistent tokens, it would probably make sense to add a test here for the second situation as well.

Reply:
I will give it a go. If I can make something reliable then I will add it. Multi-DC unit tests aren't the most stable in general.

Reply:
Done. Segfaults before and after it passes.